ndk/src/font.rs

```diff
@@ -0,0 +1,401 @@
+    /// | 200   | Extra Light (Ultra Light) | [`FontWeight::EXTRA_LIGHT`] |
+    /// | 300   | Light                     | [`FontWeight::LIGHT`]       |
+    /// | 400   | Normal (Regular)          | [`FontWeight::NORMAL`]      |
+    /// | 500   | Medium                    | [`FontWeight::MEDIUM`]      |
+    /// | 600   | Semi Bold (Demi Bold)     | [`FontWeight::SEMI_BOLD`]   |
+    /// | 700   | Bold                      | [`FontWeight::BOLD`]        |
+    /// | 800   | Extra Bold (Ultra Bold)   | [`FontWeight::EXTRA_BOLD`]  |
+    /// | 900   | Black (Heavy)             | [`FontWeight::BLACK`]       |
+    pub fn weight(&self) -> FontWeight {
+        FontWeight(unsafe { ffi::AFont_getWeight(self.ptr.as_ptr()) })
+    }
+
+    /// Return true if the current font is italic, otherwise returns false.
+    pub fn is_italic(&self) -> bool {
+        unsafe { ffi::AFont_isItalic(self.ptr.as_ptr()) }
+    }
+}
+
+#[cfg(feature = "api-level-29")]
+impl Drop for Font {
+    fn drop(&mut self) {
+        unsafe { ffi::AFont_close(self.ptr.as_ptr()) }
+    }
+}
+
+/// Corresponds to [`AFAMILY_VARIANT_*`].
+///
+/// [`AFAMILY_VARIANT_*`]: https://developer.android.com/ndk/reference/group/font#group___font_1gga96a58e29e8dbf2b5bdeb775cba46556ea662aafc7016e35d6758da93416fc0833
+#[repr(u32)]
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum FamilyVariant {
+    /// A family variant value for the compact font family variant.
+    /// The compact font family has Latin-based vertical metrics.
+    Compact = ffi::AFAMILY_VARIANT_COMPACT as _,
+    /// A family variant value for the system default variant.
+    Default = ffi::AFAMILY_VARIANT_DEFAULT as _,
+    /// A family variant value for the elegant font family variant.
+    /// The elegant font family may have larger vertical metrics than Latin font.
+    Elegant = ffi::AFAMILY_VARIANT_ELEGANT as _,
+}
+
+/// A native [`AFontMatcher *`]
+///
+/// [`AFontMatcher *`]: https://developer.android.com/ndk/reference/group/font#afontmatcher_create
+#[cfg(feature = "api-level-29")]
+#[derive(Debug)]
+pub struct FontMatcher {
+    ptr: NonNull<ffi::AFontMatcher>,
+}
+
+#[cfg(feature = "api-level-29")]
+impl FontMatcher {
+    /// Create an `FontMatcher` from a pointer
+    ///
+    /// # Safety
+    /// By calling this function, you assert that the pointer is a valid pointer to a native
+    /// `AFontMatcher`.
+    pub unsafe fn from_ptr(ptr: NonNull<ffi::AFontMatcher>) -> Self {
+        Self { ptr }
+    }
+
+    /// Returns the pointer to the native `AFontMatcher`.
+    pub fn ptr(&self) -> NonNull<ffi::AFontMatcher> {
+        self.ptr
+    }
+
+    /// Select the best font from given parameters.
+    ///
+    /// Creates a new [`FontMatcher`] object.
+    pub fn new() -> Self {
+        NonNull::new(unsafe { ffi::AFontMatcher_create() })
+            .map(|p| unsafe { FontMatcher::from_ptr(p) })
+            .expect("AFontMatcher_create returned NULL")
+    }
+
+    /// Set family variant to matcher.
```

Comment:
Bit of a crooked sentence. So `FamilyVariant` _is_ a matcher?

But `self` is already the matcher, so perhaps it meant to store/apply the argument **in/to** the matcher?
Same below.

Reply:
I usually use the form "Sets <field name> of <class name>." How about "Sets the family variant of the font to be matched."?

Reply:
BTW, I forgot to make a corresponding function of `AFontMatcher_match`! I'll make another commit for this.

Reply:
Indeed, except that there are no real field names here because it's all internal parameters. Feel free to either use your "sets" variant which is closer to Androids alien documentation, or the following suggestion which also describes what the `FontMatcher` will _end up doing_ with this information:

`Only match/return fonts with the given xxx` where `xxx` is either `locale(s)`, `language(s)` or `style`.

Or "limits", or any other sentence variant you're comfortable with.